C/services/filter-plugin-interfaces/python/python_plugin_interface.cpp

```diff
@@ -298,7 +298,7 @@ void filter_plugin_ingest_fn(PLUGIN_HANDLE handle, READINGSET *data)
 				data->removeAll();
 
 				// Append filtered readings
-				data->append(filteredReadingSet->getAllReadings());
+				data->append(filteredReadingSet);
```

Comment:
Maybe update comment also that the append call empties the passed ReadingSet also.